Add note about ng, plus light editing. (#657)
timescale/docs: api/time_bucket.md

# time_bucket()

> The `time_bucket` function is based on the standard PostgreSQL `date_trunc`

Reviewer: This is not true and very misleading. We use our own implementation, completely independent from date_trunc.

Author: How would you reword that?

Reviewer: The `time_bucket` function is somewhat similar to the standard PostgreSQL `date_trunc` function. Unlike `date_trunc`, ...

> The `time_bucket` function is based on the standard PostgreSQL `date_trunc`
> function. It allows for arbitrary time intervals instead of the second, minute,

Author: ```suggestion
The `time_bucket` function is similar to the standard PostgreSQL `date_trunc`
function. Unlike `date_trunc`, it allows for arbitrary time intervals instead of the second, minute,
```

> The `time_bucket` function is based on the standard PostgreSQL `date_trunc`
> function. It allows for arbitrary time intervals instead of the second, minute,
> and hour intervals provided by `date_trunc`. The return value is the bucket's
> start time.

Author: ```suggestion
The `time_bucket` function is similar to the standard PostgreSQL `date_trunc`
function. Unlike `date_trunc`, it allows for arbitrary time intervals instead of 
second, minute, and hour intervals. The return value is the bucket's
start time.
```

`TIMESTAMPTZ` arguments are bucketed by the time in UTC, so the alignment of
buckets is on UTC time. One consequence of this is that daily buckets are
aligned to midnight UTC, not local time. If you want buckets aligned by local
time, cast the `TIMESTAMPTZ` input to `TIMESTAMP`, which converts the value to
local time, before you pass it to `time_bucket`. For an example, see the sample
use in this section.

Note that daylight savings time boundaries means that the amount of data
aggregated into a bucket after such a cast can be irregular. For example, if the
`bucket_width` is 2 hours, the number of UTC hours bucketed by local time on
daylight savings time boundaries can be either three hours or one hour.

<highlight type="important">

> Month, year, and timezone arguments are not supported by the `time_bucket`

Reviewer: This is slightly misleading and contradicts our regular terminology. Months and/or years can be specified in the **same** argument of the function - bucket width. In other words, there are no separate arguments for months and years. Better just omit "argument" word in this context.

> Month, year, and timezone arguments are not supported by the `time_bucket`

Author: ```suggestion
Month, year, and timezones are not supported by the `time_bucket`
```
